ScheduleLesson/Controllers/ScheduleController.cs

```diff
@@ -33,19 +38,24 @@ public async Task<ActionResult<Schedule>> GetSchedule(int id)
         [HttpPost]
         public async Task<ActionResult<Schedule>> AddSchedule(Schedule schedule)
         {
-            Schedule result = await _scheduleService.AddSchedule(schedule);
+            string? userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            Schedule result = await _scheduleService.AddSchedule(schedule, userId);
             return Ok(result);
         }
+
         [HttpPut]
         public async Task<ActionResult<Schedule>> UpdateSchedule(Schedule Id)
         {
-            Schedule? dbScheduleUpdated = await _scheduleService.UpdateSchedule(Id);
+            string? userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            Schedule? dbScheduleUpdated = await _scheduleService.UpdateSchedule(Id, userId);
```

Comment:
It seems that only the id can be used here, since we can identify a unique schedule by its id.

Reply:
Id in Schedul and id is user different. If use only user id, user can edit not only their entries. Maybe I don't understand)

Reply:
Using both Id and UserId seems redundant because Id is a primary key.
Id is already unique – It uniquely identifies a record in the table, so adding UserId doesn’t improve the accuracy of update or delete operations.
Cleaner and simpler queries – Using only Id makes the code clearer and easier to maintain.

On the other hand, using both Id and UserId can enhance security.
If you want to ensure that the Id truly belongs to the specified UserId, it’s reasonable to keep this check.

For practice:
try creating a record in the schedule table with the same Id and a different UserId
try creating a record in the schedule table with a different Id and the same UserId
https://www.w3schools.com/sql/sql_primarykey.ASP

Reply:
Practice:
1. Creating a record in the schedule table with the same Id and a different UserId - it show error, because the "id" is a primary key and it cannot be repeated.
2. Creating a record in the schedule table with a different Id and the same UserId - wrote down the record in the table